### ISSUE TITLE
Ionic 4 / Angular 7 - aws amplify doesn't have Federated Sign in / Sign up

### ISSUE THREAD

**Issue by bijanmmarkes** [aws-amplify/amplify-js#2455]
**Is your feature request related to a problem? 

AWS Amplify for Ionic 4 doesn't seem to have any way to complete federated logins using amazon, google, or facebook. I have seen documentation for React Native but none for Ionic 4, How would I enable federated sign in / sign up via social logins through the `<amplify-authenticator>` ?

**Describe the solution you'd like**
I would like the ability to enable federated sign up / sign in / sign out while still being able to use the provided <amplify-authenticator> module / theme. I have seen React Native able to add a parameter to `<amplify-authenticator>` such as federated = {federated}, Why is this type of functionality not available for Ionic 4?

**Describe alternatives you've considered**
Most preferably it would be great to have the Sign up / Sign in UI come with a toggle feature for social providers, and some documentation to go with it. It seems AWS Amplify is lacking for Ionic 4 and prioritizes React, it would be nice to see some of the features React has come to Ionic 4.

**Additional context**
No way to enable federated sign in using the UI or other methods on Ionic 4 - AWS Amplify latest version.

**Comment by bijanmmarkes** [comment]
@haverchuck Any news on this?

**Comment by martin-slater** [comment]
Is there any plans for this in the roadmap? It is disappointing that it is not supported yet for ionic 4 when react has it all

**Comment by murlivarma** [comment]
any update on this?

**Comment by LiorSaadon** [comment]
any news for federated login ?

on "aws-amplify-angular": "3.0.3" i found this part in the code that might indicate federated sign in is coming, but will it support SAML ADFS?

var federatedSignIn = {
    section: 'federated-sign-in-section',
    bodySection: 'federated-sign-in-body-section',
    signInButtons: 'federated-sign-in-buttons',
};

**Comment by oliverandersencox** [comment]
This is ridiculous to not be supported in Ionic. If it's already in React Native. Currently you can use the federated sign in via the API but on mobile native app with Ionic, the redirect will go to localhost and so it breaks.

**Comment by jordanranz** [comment]
This is in progess: https://github.com/aws-amplify/amplify-js/pull/2900

**Comment by LiorSaadon** [comment]
> This is in progess: #2900

i went through the changes in the pull request, he did great job for facebook,google,amazon sign in but still no SAML ADFS :(

**Comment by oliverandersencox** [comment]
are we looking any closer on this one? Really keen to get facebook login into the app

**Comment by oliverandersencox** [comment]
no don't close this, we are waiting to here about this feature

**Comment by oliverandersencox** [comment]
@haverchuck any progress on this. We're waiting on this to release out app

**Comment by giacomocarrozzo** [comment]
Any update on this?

**Comment by oliverandersencox** [comment]
any of the aws team care to give us any sort of feedback on this?

**Comment by jordanranz** [comment]
Quick update on this. We will be implementing Federated Sign In buttons into our UI Component refactor: https://github.com/aws-amplify/amplify-js/issues/3279

**Comment by sammartinez** [comment]
Resolving this feature request as completed this in UI Components

**Comment by oliverandersencox** [comment]
for angular? if so where are the docs, as the app store REJECTS  any redirection to sfari to authenticate

**Comment by cwomack** [comment]
For anyone following this issue, Federated Sign In & Sign up is now supported with [Angular](https://ui.docs.amplify.aws/angular/connected-components/authenticator/configuration#social-providers) and [React](https://ui.docs.amplify.aws/react/connected-components/authenticator/configuration#social-providers) when using the Authenticator connected component.  

If anyone continues to experience problems with the Authenticator as described in this issue, please feel free to reply back or open a new issue within the [amplify-ui repository](https://github.com/aws-amplify/amplify-ui/issues) (where the Authenticator features/issues are now managed).
